app/components/network/networkController.js

```diff
@@ -19,12 +19,82 @@ function ($scope, $state, $stateParams, Network, Messages) {
     });
   };
 
-  $('#loadingViewSpinner').show();
-  Network.get({id: $stateParams.id}, function (d) {
-    $scope.network = d;
-    $('#loadingViewSpinner').hide();
-  }, function (e) {
-    $('#loadingViewSpinner').hide();
-    Messages.error("Failure", e, "Unable to retrieve network info");
+  $scope.containerLeaveNetwork = function containerLeaveNetwork(network, containerId) {
+    $('#loadingViewSpinner').show();
+    Network.disconnect({id: $stateParams.id}, { Container: containerId, Force: false }, function (d) {
+      if (d.message) {
+        $('#loadingViewSpinner').hide();
+        Messages.send("Error", {}, d.message);
+      } else {
+        $('#loadingViewSpinner').hide();
+        Messages.send("Container left network", $stateParams.id);
+        $state.go('network', {id: network.Id}, {reload: true});
+      }
+    }, function (e) {
+      $('#loadingViewSpinner').hide();
+      Messages.error("Failure", e, "Unable to disconnect container from network");
+    });
+  };
+
+  function getNetwork() {
+    $('#loadingViewSpinner').show();
+    Network.get({id: $stateParams.id}, function (d) {
+      $scope.network = d;
+      getContainersInNetwork(d);
+      $('#loadingViewSpinner').hide();
+    }, function (e) {
+      $('#loadingViewSpinner').hide();
+      Messages.error("Failure", e, "Unable to retrieve network info");
+    });
+  };
+
+  function getContainersInNetwork(network) {
+    if (network.Containers) {
+      Container.query({
+        filters: {network: [$stateParams.id]}
+      }, function (containersInNetworkResult) {
+        if ($scope.containersToHideLabels) {
+          containersInNetworkResult = ContainerHelper.hideContainers(containersInNetworkResult, $scope.containersToHideLabels);
+        }
+        var containersInNetwork = [];
+        containersInNetworkResult.forEach(function(container) {
+          containersInNetwork.push(network.Containers[container.Id]);
+        });
+        $scope.containersInNetwork = containersInNetwork;
+      });
+    }
+  }
+
+  function retrieveSwarmHostsInfo(data) {
+    var swarm_hosts = {};
+    var systemStatus = data.SystemStatus;
+    var node_count = parseInt(systemStatus[3][1], 10);
+    var node_offset = 4;
+    for (i = 0; i < node_count; i++) {
+      var host = {};
+      host.name = _.trim(systemStatus[node_offset][0]);
+      host.ip = _.split(systemStatus[node_offset][1], ':')[0];
+      swarm_hosts[host.name] = host.ip;
+      node_offset += 9;
+    }
+    return swarm_hosts;
+  }
+
+  $scope.swarm = false;
+  Config.$promise.then(function (c) {
+    $scope.containersToHideLabels = c.hiddenLabels;
+    $scope.swarm = c.swarm;
```

Comment:
You don't need to retrieve swarm related information as you don't use it in the HTML template. You can actually drop all the code related to setting the `$scope.swarm_hosts` and `$scope.swarm_mode` variables.

All you need here is:

```javascript
  Config.$promise.then(function (c) {
    $scope.containersToHideLabels = c.hiddenLabels;
    getNetwork();
  });
```